File: Blind-75/Array&Hashing/LongestConsecutiveSequence/Longest-Consecutive-Sequence.java
import java.util.HashSet;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        int res = 0;
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) { set.add(num);}

        for (int num : nums){
            int count = 0;
            if (!set.contains(num-1)){
                while (set.contains(num++))
                    count += 1;
            }
            res = Math.max(res, count);
        }
        return res;
    }
}

File: Blind-75/Array&Hashing/LongestConsecutiveSequence/Longest-Consecutive-Sequence.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        setNums = set(nums)
        for num in nums:
            if (num - 1) not in setNums:
                length = 1
                while (num+length) in setNums:
                    length += 1
                res = max(res, length)
        return res
